src/Microsoft.Health.Fhir.SqlServer/Features/Search/Expressions/Visitors/QueryGenerators/SqlQueryGenerator.cs

```diff
@@ -385,18 +414,62 @@ private void HandleTableKindNotExists(SearchParamTableExpression searchParamTabl
 
         private void HandleTableKindTop(SearchOptions context)
         {
-            var (paramInfo, sortOrder) = context.Sort.Count == 0 ? default : context.Sort[0];
             var tableExpressionName = TableExpressionName(_tableExpressionCounter - 1);
-            var sortExpression = (paramInfo == null || paramInfo.Code == KnownQueryParameterNames.LastUpdated) ? null : $"{tableExpressionName}.SortValue";
+            bool isSpecialCaseSort = IsPrimaryKeySort(context);
+            var sortExpression = isSpecialCaseSort ? null : "SortValue";
```

Comment:
>"SortValue" [](start = 60, length = 11)

Shoud this be `"{tableExpressionName}.SortValue"` instead of `"SortValue"` ?

Reply:
The reason I took it out is that it's not permitted in the window function `ROW_NUMBER() OVER(ORDER BY SortValue ASC, Sid1 ASC)` and it shouldn't be ambiguous in this context (there is only one CTE).